weblate/addons/models.py

```diff
@@ -216,44 +305,22 @@ def handle_addon_event(
     if addon_queryset is None:
         addon_queryset = Addon.objects.filter_event(component, event)
 
-    # Filter matching events
     for addon in addon_queryset:
-        with transaction.atomic():
-            # Extract scope from the add-on model, used for EVENT_DAILY
-            if auto_scope:
-                component = scope = addon.component
-
-            scope.log_debug("running %s add-on: %s", event.label, addon.name)
-            try:
-                # Execute event in senty span to track performance
-                with sentry_sdk.start_span(
-                    op=f"addon.{event.name}", description=addon.name
-                ):
-                    if isinstance(method, str):
-                        getattr(addon.addon, method)(*args)
-                    else:
-                        # Callback is used in tasks
-                        method(addon)
-            except DjangoDatabaseError:
-                raise
-            except Exception as error:
-                # Log failure
-                scope.log_error(
-                    "failed %s add-on: %s: %s", event.label, addon.name, error
-                )
-                report_error(
-                    cause=f"add-on {addon.name} failed", project=component.project
-                )
-                # Uninstall no longer compatible add-ons
-                if not addon.addon.can_install(component, None):
-                    scope.log_warning(
-                        "uninstalling incompatible %s add-on: %s",
-                        event.label,
-                        addon.name,
-                    )
-                    addon.disable()
-            else:
-                scope.log_debug("completed %s add-on: %s", event.label, addon.name)
+        if auto_scope:
+            component = scope = addon.component
+        elif addon.project:
+            scope = addon.project
+        elif addon.component is None and addon.project is None:
+            scope = "site-wide"  # there is a better way to handle this?
+
+        if isinstance(scope, component):
+            execute_addon_event(addon, component, scope, event, method, args)
+        if isinstance(scope, Project):
+            for _component in component.project.component_set.iterator():
+                execute_addon_event(addon, _component, scope, event, method, args)
+        else:  # site-wide
+            for _component in Component.objects.all().iterator():
+                execute_addon_event(addon, _component, scope, event, method, args)
```

Comment:
I think this should work:

```suggestion
        if not auto_scope:
            execute_addon_event(addon, component, scope, event, method, args)
        else:
            if addon.component:
                components = [addon.component]
            elif addon.project:
                components = addon.project.component_set.iterator()
            else:
                components = Component.objects.iterator()
            
            for component in components:
                execute_addon_event(addon, component, component, event, method, args)
```